### DISCUSSION TITLE
How to specific LD_LIBRARY for custom c++ toolchain?

### DISCUSSION THREAD

**Discussion by zclllyybb** [compiler-explorer/compiler-explorer#5796] [Q&A]
I set my cpp toolchain in `c++.local.properties`. and when the `make` running, it shows 
```
info: xxx/gcc Inconsistency detected by ld.so: dl-call-libc-early-init.c: 37: _dl_call_libc_early_init: Assertion `sym != NULL' failed! is ready
......
info: xxx/clang Inconsistency detected by ld.so: dl-call-libc-early-init.c: 37: _dl_call_libc_early_init: Assertion `sym != NULL' failed! is ready
```
It shows the same when I try to use them in the frontend interface. 
This error message seems a very classic one about glibc incompatibility? But this toolchain has been running very well on this machine to compile many repositories. Idk how to find the root cause.

thanks very much for any help

**Comment by zclllyybb** [comment]
maybe I should configure `LD_LIBRARY_PATH=${my_toolchain}/lib ${my_toolchain}/lib/ld-linux-x86-64.so.2` in `options`? But what's the right way

**Comment by partouf** [comment]
I don't think I've seen this error before, but we do frequently set the ldPath property in our properties files. This will automatically add the necessary LD_LIBRARY_PATH variables and pass the needed rpath's to the linker.

For example we set it here https://github.com/compiler-explorer/compiler-explorer/blob/main/etc/config/c%2B%2B.amazon.properties#L381 for our Clang compilers.

**Reply by zclllyybb** [reply]
well it still doesn't work after I set items like
```
compiler.clang_ldb_toolchain.ldPath=xxx/lib|xxx/lib/ld-linux-x86-64.so.2
```
for them. But I'll continue to try to figure out.
